Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/ContentBlockerScriptHandler.swift

```diff
@@ -63,38 +63,40 @@ extension ContentBlockerHelper: TabContentScript {
       let data = try JSONSerialization.data(withJSONObject: message.body)
       let dto = try JSONDecoder().decode(ContentBlockerDTO.self, from: data)
     
-      let isPrivateBrowsing = PrivateBrowsingManager.shared.isPrivateBrowsing
-      let domain = Domain.getOrCreate(forUrl: currentTabURL, persistent: !isPrivateBrowsing)
-      if let shieldsAllOff = domain.shield_allOff, Bool(truncating: shieldsAllOff) {
-        // if domain is "all_off", can just skip
-        return
-      }
-
-      if dto.data.resourceType == .script && domain.isShieldExpected(.NoScript, considerAllShieldsOption: true) {
-        self.stats = self.stats.adding(scriptCount: 1)
-        BraveGlobalShieldStats.shared.scripts += 1
-        return
-      }
-      
-      // Because javascript urls allow some characters that `URL` does not,
-      // we use `NSURL(idnString: String)` to parse them
-      guard let requestURL = NSURL(idnString: dto.data.resourceURL) as URL? else { return }
-      guard let sourceURL = NSURL(idnString: dto.data.sourceURL) as URL? else { return }
-      guard let domainURLString = domain.url else { return }
-      
-      // Getting this domain and current tab urls before going into asynchronous closure
-      // to avoid threading problems(#1094, #1096)
-      assertIsMainThread("Getting enabled blocklists should happen on main thread")
-      let loadedRuleTypes = Set(loadedRuleTypeWithSourceTypes.map({ $0.ruleType }))
-
-      TPStatsBlocklistChecker.shared.isBlocked(
-        requestURL: requestURL,
-        sourceURL: sourceURL,
-        loadedRuleTypes: loadedRuleTypes,
-        resourceType: dto.data.resourceType
-      ) { blockedType in
+      Task { @MainActor in
+        let isPrivateBrowsing = PrivateBrowsingManager.shared.isPrivateBrowsing
+        let domain = Domain.getOrCreate(forUrl: currentTabURL, persistent: !isPrivateBrowsing)
+        if let shieldsAllOff = domain.shield_allOff, Bool(truncating: shieldsAllOff) {
+          // if domain is "all_off", can just skip
+          return
+        }
+        
+        if dto.data.resourceType == .script && domain.isShieldExpected(.NoScript, considerAllShieldsOption: true) {
+          self.stats = self.stats.adding(scriptCount: 1)
+          BraveGlobalShieldStats.shared.scripts += 1
+          return
+        }
+        
+        // Because javascript urls allow some characters that `URL` does not,
+        // we use `NSURL(idnString: String)` to parse them
+        guard let requestURL = NSURL(idnString: dto.data.resourceURL) as URL? else { return }
+        guard let sourceURL = NSURL(idnString: dto.data.sourceURL) as URL? else { return }
+        guard let domainURLString = domain.url else { return }
+        
+        // Getting this domain and current tab urls before going into asynchronous closure
+        // to avoid threading problems(#1094, #1096)
+        assertIsMainThread("Getting enabled blocklists should happen on main thread")
```

Comment:
This can probably be removed now

Reply:
Fixed here: https://github.com/brave/brave-ios/pull/6507/commits/1cde37c472b2965851f1692e23ec27727b0e7f45